src/main/java/org/springframework/data/jpa/repository/support/CrudMethodMetadataPostProcessor.java

```diff
@@ -226,25 +227,25 @@ private static LockModeType findLockModeType(Method method) {
 			return annotation == null ? null : (LockModeType) AnnotationUtils.getValue(annotation);
 		}
 
-		private static Map<String, Object> findQueryHints(Method method, Predicate<QueryHints> annotationFilter) {
+		private static List<QueryHintValue> findQueryHints(Method method, Predicate<QueryHints> annotationFilter) {
```

Comment:
Let's collect hints in a `MultiValueMap`. That way, we can avoid the additional type.